components/admin_console/schema_admin_settings.jsx

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+import {FormattedHTMLMessage, FormattedMessage} from 'react-intl';
+
+import * as I18n from 'i18n/i18n.jsx';
+
+import {SettingsTypes} from 'utils/constants.jsx';
+import {formatText} from 'utils/text_formatting.jsx';
+import * as Utils from 'utils/utils.jsx';
+import RequestButton from 'components/admin_console/request_button/request_button';
+import LoadingScreen from 'components/loading_screen.jsx';
+import AdminSettings from 'components/admin_console/admin_settings.jsx';
+import BooleanSetting from 'components/admin_console/boolean_setting.jsx';
+import TextSetting from 'components/admin_console/text_setting.jsx';
+import DropdownSetting from 'components/admin_console/dropdown_setting.jsx';
+import MultiSelectSetting from 'components/admin_console/multiselect_settings.jsx';
+import RadioSetting from 'components/admin_console/radio_setting.jsx';
+import GeneratedSetting from 'components/admin_console/generated_setting.jsx';
+import UserAutocompleteSetting from 'components/admin_console/user_autocomplete_setting.jsx';
+import SettingsGroup from 'components/admin_console/settings_group.jsx';
+
+export default class SchemaAdminSettings extends AdminSettings {
+    componentWillReceiveProps(nextProps) {
+        if (nextProps.schema !== this.props.schema) {
+            this.setState(this.getStateFromConfig(nextProps.config, nextProps.schema));
+        }
+    }
+
+    getConfigFromState(config) {
+        const schema = this.props.schema;
+
+        if (schema) {
+            if (!config[schema.id]) {
+                config[schema.id] = {};
+            }
+
+            const configSettings = config[schema.id];
+
+            const settings = schema.settings || [];
+            settings.forEach((setting) => {
+                configSettings[setting.key] = this.state[setting.key];
+            });
```

Comment:
The parent class is AdminSetting, so it receive the global config from there. Once I have the config, i get the configSettings as a "pointer" to the config[schema.id] object in the config data structure. After that, I change the attributes of that object (actually I'm changing the config data under the hood, changing the an object contained in config object).

Reply:
Thanks for the explanation! 
(I'm hoping that whenever possible, we should be using pure function as much as possible.)

Reply:
I can refactor it a bit to give pure behavior.

Reply:
That would be great! Thanks!

Reply:
I was concernet about modify config from the store directly, but actually in the AdminSetting component we "clone" the entire field to avoid that using `JSON.parse(JSON.stringify(this.props.config))` before call getConfigFromState, so I think the behavior is "almost" pure. We can move that into the getConfigFromState, but I'm not sure is a good idea, because remove the "general" protection.